README.md

```diff
@@ -237,22 +237,26 @@ If the `ConfigMap` is not disabled, then upon start, CCM does the following:
-   * Service addition: create a facility-specific `/32` IP address reservation tagged with `usage="packet-ccm-auto"` and `service="<service-hash>"`; if it is ready immediately, add it to the [service spec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#servicespec-v1-core) and ensure is in the pools of the metallb `ConfigMap` with `auto-assign: false`
+   * Service addition: create a facility-specific `/32` IP address reservation tagged with appropriate tags; if it is ready immediately, add it to the [service spec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#servicespec-v1-core) and ensure is in the pools of the metallb `ConfigMap` with `auto-assign: false`
    * Service deletion: find the `/32` IP address on the service spec and remove it; remove from the `ConfigMap`
-1. Start an independent loop that checks every 30 seconds (configurable) for IP address reservations that are tagged with `usage="packet-ccm-auto"` and `service="<service-hash>"` but not on any services. If it finds one:
+1. Start an independent loop that checks every 30 seconds (configurable) for IP address reservations that are tagged with appropriate tags but not on any services. If it finds one:
    * If a service exists that matches the `<service-hash>`, that is an indication that an IP address reservation request was made, not completed at request time, and now is available. Add the IP to the [service spec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#servicespec-v1-core) and ensure is in the pools of the metallb `ConfigMap` with `auto-assign: false`
    * If no service exists that is missing an IP, or none with a matching hash, delete the IP reservation
 
 At **no** point does CCM itself deploy the load-balancer or any part of it, including the `ConfigMap`. It only
 modifies an existing `ConfigMap`. This can be deployed by the administrator separately, using the manifest
 provided in the releases page, or by any other manner.
 
-In all cases of tagging the IP address reservation, we tag the IP reservation with `usage="packet-ccm-auto"` and `service="<service-hash>"` where `<service-hash>` is the sha256 hash of `<namespace>.<service-name>`. We do this so that the name of the service does not leak out to Equinix Metal itself.
+In all cases of tagging the IP address reservation, we tag the IP reservation with the following tags:
 
-IP addresses always are created `/32`, and are tagged as `"packet-ccm-auto"`
+* `usage="packet-ccm-auto"`
+* `service="<service-hash>"` where `<service-hash>` is the sha256 hash of `<namespace>/<service-name>`. We do this so that the name of the service does not leak out to Equinix Metal itself.
+* `cluster=<clusterID>` where `<clusterID>` is the UID of the immutable `kube-system` namespace. We do this so that if someone runs two clusters in the same project, and there is one `Service` in each cluster with the same namespace and name, then the two EIPs will not conflict.
```

Comment:
It looks like the other ccm implementations use a command line argument/config to make this configurable by the user (this is also something that we could template out pretty easily in cluster-api when deploying the CCM), this would prevent potential issues related to the UID changing if the cluster is restored from backup for any reason.

It looks like there is a shared configuration that is exposed through k8s.io/cloud-provider, which is what is used by the legacy ccm and the aws out-of-tree provider: https://github.com/kubernetes/cloud-provider/blob/master/config/v1alpha1/types.go#L60

It looks like vsphere went a different route and exposed their own flag: https://github.com/kubernetes/cloud-provider-vsphere/blob/22e85632bfe62f9a6afff8d2fc269f80eb9e7c78/cmd/vsphere-cloud-controller-manager/main.go

Reply:
@detiber these are good reference points; thanks for bringing them here.

As I look through the [cloudprovider.Interface](https://godoc.org/k8s.io/cloud-provider#Interface), I see that the following all have methods that are passed a `clusterName string`:

* [Clusters](https://godoc.org/k8s.io/cloud-provider#Clusters)
* [LoadBalancer[(https://godoc.org/k8s.io/cloud-provider#LoadBalancer)
* [Routes](https://godoc.org/k8s.io/cloud-provider#Routes)

Are those the same [ClusterName](https://github.com/kubernetes/cloud-provider/blob/master/config/v1alpha1/types.go#L60) you linked to above, from the cloud-provider types? How is it set and used? It says [in the comment](https://github.com/kubernetes/cloud-provider-vsphere/blob/49f6445da72ec3c607125aa9aee89c8d830a8754/pkg/cloudprovider/vsphere/loadbalancer/lbprovider.go#L121):

```go
// Parameter 'clusterName' is the name of the cluster as presented to kube-controller-manager
```

Is that set via the `--cluster-name` parameter set on [kube-controller-manager](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/), which defaults to `"kubernetes"`?

Are you suggesting we just use that as the cluster name, and then let the creator decide on what name they want to use, and inherit it?